lib/Core/SQLiteBuildDB.cpp

```diff
@@ -28,16 +29,28 @@ using namespace llbuild::core;
 
 // SQLite BuildDB Implementation
 
-// FIXME: This entire implementation needs to be updated for good error
-// handling.
-
 namespace {
 
 class SQLiteBuildDB : public BuildDB {
   static const int currentSchemaVersion = 5;
 
   sqlite3 *db = nullptr;
 
+  std::string getCurrentErrorMessage() {
+    int err_code = sqlite3_errcode(db);
+    const char* err_message = sqlite3_errmsg(db);
+    const char* filename = sqlite3_db_filename(db, "main");
+
+    std::stringstream out;
+    out << "error: accessing build database \"" << filename << "\": " << err_message;
+
+    if (err_code == SQLITE_BUSY || err_code == SQLITE_LOCKED) {
+      out << "\nPossibly there are two concurrent builds running in the same filesystem location.";
```

Comment:
Let's make this into one line... it's not as pretty, but it matches the expectations of higher level tools better until we have "real" diagnostic infrastructure.